trinity/rpc/modules/eth.py

```diff
@@ -264,5 +267,12 @@ class Eth(RPCModule):
     async def protocolVersion(self) -> str:
         return "63"
 
-    async def syncing(self) -> bool:
-        raise NotImplementedError()
+    async def syncing(self) -> Union[bool, Dict[str, BlockNumber]]:
```

Comment:
For stronger type safety we've started using `mypy_extensions.TypedDict` for these types of dictionaries with known keys and value types.

https://github.com/python/mypy/blob/a7f0263f29317985e0a2ca6f85ba5b7666401347/extensions/mypy_extensions.py#L74-L99